src/tools/codegen-unit-test.ts

```diff
@@ -0,0 +1,137 @@
+/**
+ * @license
+ * Copyright (c) 2019 Google Inc. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import fs from 'fs';
+import {Manifest} from '../runtime/manifest.js';
+import {Flags} from '../runtime/flags.js';
+
+type Test = {
```

Comment:
It seems like you often prefer to create a `type` instead of an `interface` (equivalent). Out of curiosity, why one syntax over the other?

Reply:
They seem to be _mostly_ equivalent:
https://medium.com/@martin_hotell/interface-vs-type-alias-in-typescript-2-7-2a8f1777af4c

What's your take on this?

Reply:
I like using `type` for unions of other types, for aliases of one type (like a lambda), or with this [`Readonly` trick](https://github.com/PolymerLabs/arcs/blob/master/src/tools/schema2cpp.ts#L127).

I think if we're describing an object with multiple fields, an interface is more appropriate. It's certainly just a style preference. But this is what seems appropriate in my head.